File: app/src/main/java/com/ireddragonicy/konabessnext/core/strategy/ChipArchitecture.java
package com.ireddragonicy.konabessnext.core.strategy;

import java.util.List;
import com.ireddragonicy.konabessnext.model.Bin;

public interface ChipArchitecture {
    /**
     * Checks if the current line indicates the start of a GPU power level block.
     */
    boolean isStartLine(String line);

    /**
     * Decodes lines from the DTS file into Bin objects.
     * 
     * @param dtsLines   The full list of lines in the DTS file.
     * @param bins       The list to populate with parsed Bin objects.
     * @param startIndex The line index where the block starts.
     */
    void decode(List<String> dtsLines, List<Bin> bins, int startIndex) throws Exception;

    /**
     * Generates DTS string lines from the Bin objects.
     */
    List<String> generateTable(List<Bin> bins);
}

File: app/src/main/java/com/ireddragonicy/konabessnext/core/strategy/ChipArchitecture.kt
package com.ireddragonicy.konabessnext.core.strategy

import com.ireddragonicy.konabessnext.model.Bin

interface ChipArchitecture {
    /**
     * Checks if the current line indicates the start of a GPU power level block.
     */
    fun isStartLine(line: String): Boolean

    /**
     * Decodes lines from the DTS file into Bin objects.
     *
     * @param dtsLines   The full list of lines in the DTS file.
     * @param bins       The list to populate with parsed Bin objects.
     * @param startIndex The line index where the block starts.
     */
    @Throws(Exception::class)
    fun decode(dtsLines: MutableList<String>, bins: MutableList<Bin>, startIndex: Int)

    /**
     * Generates DTS string lines from the Bin objects.
     */
    fun generateTable(bins: List<Bin>): List<String>
}
